src/test/article.spec.js

```diff
@@ -668,3 +677,137 @@ describe('PUT /api/v1/articles/:articleId', () => {
+    expect(response.body.message).to.equal('articleId must be a number');
+  });
+  it('Returns 400 if the title provided is less than 3 characters', async () => {
+    const firstUser = await createUser();
+    const token = await helper.jwtSigner(firstUser);
+
+    const response = await server()
+      .put(`${ARTICLES_API}/publish`)
+      .set('x-access-token', token)
+      .send(articleWithShortTitle);
+    expect(response.status).to.equal(400);
+    expect(response.body.message).to.equal(
+      'Enter a Title for your Article of at least 3 characters'
+    );
+  });
+  it('Returns 400 if the body of the article provided is less than 3 characters', async () => {
+    const firstUser = await createUser();
+    const token = await helper.jwtSigner(firstUser);
+
+    const response = await server()
+      .put(`${ARTICLES_API}/publish`)
+      .set('x-access-token', token)
+      .send(articleWithShortBody);
+    expect(response.status).to.equal(400);
+    expect(response.body.message).to.equal('Enter the body of your Article');
+  });
+  it('Returns 400 if the articles image provided not a url', async () => {
+    const firstUser = await createUser();
+    const token = await helper.jwtSigner(firstUser);
+
+    const response = await server()
+      .put(`${ARTICLES_API}/publish`)
+      .set('x-access-token', token)
+      .send(articleWithShortImage);
+    expect(response.status).to.equal(400);
+    expect(response.body.message).to.equal('Image must contain a valid url');
+  });
+  it('Returns 400 if the articles description was not provided', async () => {
+    const firstUser = await createUser();
+    const token = await helper.jwtSigner(firstUser);
+
+    const response = await server()
+      .put(`${ARTICLES_API}/publish`)
+      .set('x-access-token', token)
+      .send(articleWithShortDescription);
+    expect(response.status).to.equal(400);
+    expect(response.body.message).to.equal(
+      'Enter a brief description for the article'
+    );
+  });
+  it('Returns 200 if article saved as draft has been published', async () => {
+    const firstUser = await createUser();
+    const token = await helper.jwtSigner(firstUser);
+    const articles = await generateArticle({ authorId: firstUser.id });
+    const createdArticle = await createArticle(articles);
+
+    const response = await server()
+      .put(`${ARTICLES_API}/publish?articleId=${createdArticle.id}`)
+      .set('x-access-token', token)
+      .send(article);
+    expect(response.status).to.equal(200);
+    expect(response.body.data.title).to.equal(article.title);
+    expect(response.body.data.body).to.equal(article.body);
+    expect(response.body.data.image).to.equal(article.image);
+    expect(response.body.data.authorId).to.equal(articles.authorId);
+    expect(response.body.data.description).to.equal(article.description);
+  });
+  it('Returns 200 if article saved as draft has is been published multiple times and the ublished date is not changing', async () => {
```

Comment:
You can change the title to => `Publishing an article multiple times fails silently`